### PR TITLE
feat(1967): Add ability to do raw query. BREAKING CHANGE: throws err when query function not implemented

### PR DESCRIPTION
**Pull request by InderH**
## Context

Need to add the ability to do raw queries to datastore so queries that are otherwise impossible to do through sequelize can be done.

## Objective

Add a query method to datastore.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1967

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by klu909 — APPROVED** [review]

**Review by jithine — APPROVED** [review]

File: test/index.test.js [hunk 1]
```diff
@@ -199,6 +207,36 @@ describe('index test', function () {
...
+            instance.query({})
+                .then(() => {
+                    throw new Error('Oops');
+                }, (err) => {
+                    assert.isOk(err, 'Error should be returned');
+                    assert.equal(err.name, 'ValidationError');
+                })
+        );
+
+        it('returns error if config object is valid and _scan not overridden', () => {
```
Comment (tkyi) [inline test/index.test.js]:
```suggestion
        it('returns error if config object is valid and _query not overridden', () => {
```

File: index.js [hunk 2]
```diff
@@ -112,6 +112,24 @@ class Datastore {
...
+     * @param  {Object}        [config.replacements] Parameters to replace in the query
+     * @param  {Boolean}       [config.rawResponse]  Return raw response without binding to model
+     */
+    query(config) {
+        return validate(config, datastoreSchema.query)
+            .then(validConfig => this._query(validConfig));
+    }
+
+    _query() {
+        return Promise.reject(new Error('Not implemented'));
```
Comment (tkyi) [inline index.js:130]: If we're going to throw error if not defined, should probably bump major version

**Comment by InderH** [comment]
Whoops, added to the wrong readme.
@tkyi are these readme changes also needed or should I revert?

Edit:
Actually, looks like you linked me to this repos read me in https://github.com/screwdriver-cd/data-schema/pull/391, was that intended?

**Comment by tkyi** [comment]
@InderH probably an accident. Just need a README update in all repos where relevant
so not in datastore-sequelize, but yes in this one: https://github.com/screwdriver-cd/datastore-base/blob/master/README.md

File: index.js (hunk 2 above)
Comment (tkyi) [reply index.js:130]: Updated PR description to reflect this^

**Comment by tkyi** [comment]
One small suggestion above otherwise looks good to me

**Review by tkyi — APPROVED** [review]

**Pull request merged**